### ISSUE TITLE
Error in computation of bearing capacity for axial loading

### ISSUE THREAD

**Issue by DanielGMorenaFhecor** [fib-international/structuralcodes#140]
When computing the bearing capacity of a cross-section with an axial load N ≠ 0, the results are incorrect at first instance. Comparing these results with established software reveals discrepancies. It works correctly when the center of gravity is at the origin.  

See Point 6 in [this notebook](https://github.com/fib-international/structuralcodes/blob/a48150c72c59b20f3fa8d481ee5191e1b3cdf67e/Issues.ipynb) developed by @MestreCarlos for more information and illustrations.

**Comment by talledodiego** [comment]
I still need to dig into it and I will do as soon as I am back from vacation.
From what you say it works when center of gravity is at the origin. 
This is an intended behavior: the moments are computed integrating the normal stresses respect to global Y and Z axis. When you have no external axial force it is independent on the coordinates of the center of gravity. When there is external axial force there is also the offset moment due to N * e. It is like having a beam element with a rigid offset at the end.

This is why in our examples generally we translate the section like in this example:

```
# imports
from math import pi
from shapely import Polygon
from structuralcodes.materials.concrete import ConcreteEC2_2004
from structuralcodes.materials.reinforcement import ReinforcementEC2_2004
from structuralcodes.geometry import SurfaceGeometry
from structuralcodes.sections import add_reinforcement_line, GenericSection

# Materials
concrete = ConcreteEC2_2004(25)
reinforcemnet = ReinforcementEC2_2004(fyk=500, Es=200000, density=7850, ftk=550, epsuk=0.07)  
# Create section
poly = Polygon(((0, 0), (350, 0), (350, 500), (0, 500)))
geo = SurfaceGeometry(poly, concrete)
geo = add_reinforcement_line(geo, (50, 450), (300, 450), 20, reinforcemnet, n=10)
geo = add_reinforcement_line(geo, (50, 50), (300, 50), 20, reinforcemnet, n=10)
# Without axial force this is actually not needed, but I am doing anyway
# because it changes when there is axial force.
geo = geo.translate(-175, -250)
sec = GenericSection(geo)

sec.geometry
```

In some cases the user may want to account for this rigid offset.

**Comment by DanielGMorenaFhecor** [comment]
Thanks for your response! I've updated the example to translate the section as you suggested. I'll leave the updated example (Point 6 in [this](https://github.com/fib-international/structuralcodes/blob/961fd22c9f6ea8ee4db4dd82773fdc2ddf64d83c/Issues.ipynb) new update) and the results here, and we can discuss them further when you're back. 

These results have been updated with your latest fix to Marin's algorithm in PR #133.

```python 
# Materials
concrete = ConcreteEC2_2004(25)
reinforcement = ReinforcementEC2_2004(fyk=500, Es=200000, density=7850, ftk=550, epsuk=0.07)  

# Create section
poly = Polygon(((0, 0), (350, 0), (350, 500), (0, 500)))
geo = SurfaceGeometry(poly, concrete)
geo = add_reinforcement_line(geo, (50, 50), (300, 50), 12, reinforcement, n=3)
geo = add_reinforcement_line(geo, (50, 450), (300, 450), 20, reinforcement, n=6)
geo = geo.translate(-175, -250)  # Translate the section
sec = GenericSection(geo)

# Compute bending strength for a list of axial loads
for n_ed in [-3500, -3000, -2500, -2000, -1500, -1000, -500, 0, 500, 900]:
    m_y = sec.section_calculator.calculate_bending_strength(n=n_ed*1e3).m_y / 1e6
    print(f"N={n_ed:.2f} N\t M={m_y:.2f} kNm")
```
Output:
```
N=-3500.00 N  M=-207.35 kNm
N=-3000.00 N  M=-289.86 kNm
N=-2500.00 N  M=-343.26 kNm
N=-2000.00 N  M=-369.23 kNm
N=-1500.00 N  M=-340.02 kNm
N=-1000.00 N  M=-262.75 kNm
N=-500.00 N   M=-166.01 kNm
N=0.00 N      M=-65.78 kNm
N=500.00 N    M=35.05 kNm
N=900.00 N    M=118.06 kNm
```

To summarize, here’s the image with the results from the software we're using (check the table in the middle of the image):

![Results Image](https://github.com/fib-international/structuralcodes/blob/a48150c72c59b20f3fa8d481ee5191e1b3cdf67e/error_N_M.png?raw=true)

**Comment by talledodiego** [comment]
I am not sure I am following the issue.
To me it seems fine:

```
# imports
from math import pi
from shapely import Polygon
import numpy as np
import matplotlib.pyplot as plt
from structuralcodes.materials.concrete import ConcreteEC2_2004
from structuralcodes.materials.reinforcement import ReinforcementEC2_2004
from structuralcodes.geometry import SurfaceGeometry
from structuralcodes.sections import add_reinforcement_line, GenericSection

# Materials
concrete = ConcreteEC2_2004(25)
reinforcemnet = ReinforcementEC2_2004(fyk=500, Es=200000, density=7850, ftk=550, epsuk=0.07)  
reinforcemnet.constitutive_law = 'elasticperfectlyplastic'
# Create section
poly = Polygon(((0, 0), (350, 0), (350, 500), (0, 500)))
geo = SurfaceGeometry(poly, concrete)
geo = add_reinforcement_line(geo, (50, 450), (300, 450), 20, reinforcemnet, n=6)
geo = add_reinforcement_line(geo, (50, 50), (300, 50), 12, reinforcemnet, n=3)
# Without axial force this is actually not needed, but I am doing anyway
geo = geo.translate(-175, -250)
sec = GenericSection(geo)
# sec = GenericSection(geo, integrator='fiber', mesh_size=0.0001)

sec.geometry
```
![image](https://github.com/user-attachments/assets/8fd383f8-e970-47cd-a960-6268ad3c9cc3)

I try using interaction domain instead of bending strength calculator:
```
res_domain_0 = sec.section_calculator.calculate_nm_interaction_domain()
res_domain_180 = sec.section_calculator.calculate_nm_interaction_domain(theta=pi)
plt.plot(res_domain_0.n*1e-3, res_domain_0.m_y*1e-6)
plt.plot(res_domain_180.n*1e-3, res_domain_180.m_y*1e-6)
N = np.array([-3500, -3000, -2500, -2000, -1500, -1000, -500, 0, 500, 900])
N = N * 1e3
idx_ord = np.argsort(res_domain_0.n)
My = np.interp(N, res_domain_0.n[idx_ord], res_domain_0.m_y[idx_ord])
for n_i, my_i in zip(N, My):
    print(f'N = {n_i:5}\tkN\t\tMy = {my_i*1e-6:7.1f}\tkNm')
    plt.plot(n_i*1e-3, my_i*1e-6, 'o')
plt.grid()
plt.xlabel('N [kN]')
plt.ylabel('My [kNm]')
plt.plot(-N_vcaslu, -M_vcaslu, '--o', markersize=2, lw=0.5, label='VCA-slu')
plt.legend()
```
which returns your same numbers:
```
N = -3500000.0	kN		My =  -207.3	kNm
N = -3000000.0	kN		My =  -289.9	kNm
N = -2500000.0	kN		My =  -343.1	kNm
N = -2000000.0	kN		My =  -369.2	kNm
N = -1500000.0	kN		My =  -339.8	kNm
N = -1000000.0	kN		My =  -262.7	kNm
N = -500000.0	kN		My =  -166.0	kNm
N =   0.0	kN		My =   -65.8	kNm
N = 500000.0	kN		My =    35.1	kNm
N = 900000.0	kN		My =   118.1	kNm
```
![image](https://github.com/user-attachments/assets/fbb78357-fac0-4a9a-afc6-05b7cdff2fbf)

Indeed using bending strength calculator:
```
m_bs = []

for n_i in N: 
    res = sec.section_calculator.calculate_bending_strength(theta=0, n=n_i)
    m_bs.append(res.m_y)
    print(f'N = {n_i:5}\tkN\t\tMy = {res.m_y*1e-6:7.1f}\tkNm')
m_bs = np.array(m_bs)
```
we obtain:
```
N = -3500000.0	kN		My =  -207.4	kNm
N = -3000000.0	kN		My =  -289.9	kNm
N = -2500000.0	kN		My =  -343.3	kNm
N = -2000000.0	kN		My =  -369.2	kNm
N = -1500000.0	kN		My =  -340.0	kNm
N = -1000000.0	kN		My =  -262.8	kNm
N = -500000.0	kN		My =  -166.0	kNm
N =   0.0	kN		My =   -65.8	kNm
N = 500000.0	kN		My =    35.1	kNm
N = 900000.0	kN		My =   118.1	kNm
```

That are the same numbers.
To see a graphical comparison:

```
plt.plot(res_domain_0.n*1e-3, res_domain_0.m_y*1e-6)
plt.plot(res_domain_180.n*1e-3, res_domain_180.m_y*1e-6)
N = np.array([-3500, -3000, -2500, -2000, -1500, -1000, -500, 0, 500, 900])
N = N * 1e3
idx_ord = np.argsort(res_domain_0.n)
My = np.interp(N, res_domain_0.n[idx_ord], res_domain_0.m_y[idx_ord])
plt.plot(N*1e-3, My*1e-6, 'o', label='nm_domain')
plt.plot(N*1e-3, m_bs*1e-6, 's', markersize=10, markerfacecolor='none', label='bend_strength')

plt.grid()
plt.xlabel('N [kN]')
plt.ylabel('My [kNm]')
plt.plot(-N_vcaslu, -M_vcaslu, '--o', markersize=2, lw=0.5, label='VCA-slu')
plt.legend()
```
which returns:

![image](https://github.com/user-attachments/assets/44e9095c-9dae-4d80-8e26-8650c93f6c4d)


Note that VCA-slu is a standard software for RC sections used in Italy developed by Prof. Gelfi of University of Bologna. I did not paste here the arrays for making the plots, but I can paste them if needed.

**Comment by mortenengen** [comment]
Is this issue solved by translating the section to the origin?
